src/runtime/internal/dev.ts

```diff
@@ -256,11 +256,47 @@ export class SvelteComponentTyped<
 	 */
 	$$slot_def: Slots;
 
-	constructor(options: IComponentOptions<Props>) {
+	constructor(options: ComponentConstructorParams<Props>) {
 		super(options);
 	}
 }
 
+/**
+ * Convenience type to get the type of a Svelte component. Useful for example in combination with
+ * dynamic components and `<svelte:element>`.
+ * 
+ * Example:
+ * ```html
+ * <script lang="ts">
+ * 	import type { ComponentType, SvelteComponentTyped } from 'svelte';
+ * 	import Component1 from './Component1.svelte';
+ * 	import Component2 from './Component2.svelte';
+ * 
+ * 	const component: ComponentType = someLogic() ? Component1 : Component2;
+ * 	const componentOfCertainSubType: ComponentType<SvelteComponentTyped<{ needsThisProp: string }>> = someLogic() ? Component1 : Component2;
+ * </script>
+ * 
+ * <svelte:element this={component} />
+ * <svelte:element this={componentOfCertainSubType} needsThisProp="hello" />
+ * ```
+ */
+export type ComponentType<T extends SvelteComponentTyped = SvelteComponentTyped<any, any, any>> =
+	new (p: ComponentConstructorParams<T extends SvelteComponentTyped<infer X, any, any> ? X : any>) => T;
+
+ * Convenience type to get the props the given component expects. Example:
+ * ```html
+ * <script lang="ts">
+ * 	import type { ComponentProps } from 'svelte';
+ * 	import Component from './Component.svelte';
+ * 
+ * 	const props: ComponentProps<Component> = { foo: 'bar' }; // Errors if these aren't the correct props
+ * </script>
+ * ```
+ */
+export type ComponentProps<T extends SvelteComponent> = T extends SvelteComponentTyped<infer Props>
```

Comment:
I would probably use this the most so it's really helpful. To discuss further on your other questions, I think the simple docs for this one is good enough for now until we have a dedicated TS section like SvelteKit.

The other one is the name, currently we have 2 types (or classes) exported from the main `'svelte'` module and both are prefixed with `Svelte...`, should we also prefix this one for consistency? I do prefer not having it since it's simpler and importing from `'svelte'` should already implies that this is a svelte specific.

Reply:
Had the same thoughts and landed on the same conclusion as you - prefixing it all with Svelte just seems too verbose.